### PR TITLE
kafka: Log unknown / unsupported API keys when denying requests

### PR DESCRIPTION
**Pull request by rlenglet**
Signed-off-by: Romain Lenglet <romain@covalent.io>

### CODE REVIEW

**Comment by rlenglet** [comment]
test-me-please

File: pkg/kafka/request.go [hunk 1]
```diff
@@ -142,12 +141,6 @@ func offsetFetchTopics(req *proto.OffsetFetchReq) []string {
 // message. The response will have the specified error code set in all topics
 // and embedded partitions.
 func (req *RequestMessage) CreateResponse(err error) (*ResponseMessage, error) {
-	if req == nil || req.request == nil {
-		return nil, fmt.Errorf("request is nil")
-	}
-
-	// FIXME: Send response versions based on request
```
Comment (tgraf) [inline pkg/kafka/request.go:149]: Created https://github.com/cilium/cilium/issues/2883 to track this

**Review by tgraf — APPROVED** [review]

**Review by manalibhutiyani — COMMENTED** [review]
looks good

**Comment by manalibhutiyani** [comment]
@rlenglet  This logs the unknown key, but does not really fix the issue? We might be missing a message which needs to be whitelisted for produce/consume like other messages(which we currently hardcode). this might need changes in the Kafka parser. and then vendoring

**Pull request merged**